sqlmesh/core/model/cache.py

```diff
@@ -44,22 +42,20 @@ def get_or_load(self, name: str, entry_id: str = "", *, loader: t.Callable[[], M
             The model definition.
         """
         cache_entry = self._file_cache.get(name, entry_id)
-        if cache_entry:
-            model = cache_entry.model
-            model._full_depends_on = cache_entry.full_depends_on
-            return model
+        if isinstance(cache_entry, _Model):
+            return cache_entry
 
-        loaded_model = loader()
-        if isinstance(loaded_model, SqlModel):
-            new_entry = SqlModelCacheEntry(
-                model=loaded_model, full_depends_on=loaded_model.full_depends_on
-            )
-            self._file_cache.put(name, entry_id, value=new_entry)
+        model = loader()
+        if isinstance(model, SqlModel):
+            # make sure we preload full_depends_on
+            model.full_depends_on
```

Comment:
Such a hack :) This is why `full_depends_on` was a separate attribute - to explicitly store it and not rely on internal mechanics of materializing the cache.